forte/data/multi_pack.py

```diff
@@ -176,6 +176,172 @@ def get_span_text(self, begin: int, end: int):
+        in order to keep the index for the packs intact
+
+        Args:
+            pack (DataPack): The existing data pack.
+            index_of_pack: the index of the pack to be removed
+            clean_invalid_entries: Switch for automatically
+              cleaning the entries associated with the data pack
+              being deleted which will become invalid after the
+              removal of the pack. Default is False.
+
+        Returns:
+            True if successful
+
+        Exceptions:
+            if clean_invalid_entries is set to False and the DataPack to be removed have
+            entries (in links, groups) associated with it, ValueError will be raised.
+        """
+
+        # check if the pack to be removed has any cross pack links/groups
+        links_with_pack_for_removal = []
+        link: MultiPackLink
+        for link in self.get(MultiPackLink):
+            parent_entry_pid = link.get_parent().pack_id
+            child_entry_pid = link.get_child().pack_id
+            if (
+                parent_entry_pid == pack.pack_id
+                or child_entry_pid == pack.pack_id
+            ):
+                links_with_pack_for_removal.append(link)
+
+        groups_with_pack_for_removal = []
+        g: MultiPackGroup
+        for g in self.get(MultiPackGroup):
+            # e: Annotation
+            for e in g.get_members():
+                if e.pack_id == pack.pack_id:
+                    groups_with_pack_for_removal.append(g)
+
+        if (
+            len(links_with_pack_for_removal) > 0
+            or len(groups_with_pack_for_removal) > 0
+        ):
+            if clean_invalid_entries:
+                # clean links and groups
+                for link in links_with_pack_for_removal:
+                    # delete_entry will take care of related indexes
+                    self.delete_entry(link)
+                for g in groups_with_pack_for_removal:
+                    # delete_entry will take care of related indexes
+                    self.delete_entry(g)
+            else:  # raise exception according to requirement
+                raise ValueError(
+                    "The pack to be removed has cross-pack references."
+                    " Please set clean_invalid_entries to be True to "
+                    " auto-remove all references to this pack"
+                )
+
+        # To keep the remaining element 's index unchanged, set to None in
+        # place instead of direct removal
+
+        # vin = int(0)
+        # vin = None
+        self._pack_ref[index_of_pack] = None  # type: ignore
+        # remove(pack.pack_id) in case don't care index change
+
+        # Remove the reverse mapping from pack id to the pack index.
+        self._inverse_pack_ref.pop(pack.pack_id)
+
+        # Remove the pack names. To keep the remaining element's index
+        # unchanged, set to None instead of direct removal
+        tmp_pack_name = self.pack_names[index_of_pack]
```

Comment:
Since `tmp_pack_name` is only used once, probably we should call it inline at line 307